### PR TITLE
a scenario on contract keys

### PR DESCRIPTION
**Pull request by ghost**
A scenario on contract keys. You can try it out [here](https://github.com/digital-asset/daml-katacoda/pull/76).

### CODE REVIEW

**Review by nemanja-da — APPROVED** [review]
Awesome, it works!

File: fundamental-concepts/contract-keys/1_chasing_contract_ids.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+If your application maintains state, like a counter, or the parties following a user in the social
+network of `create-daml-app`, you will need store that state in a contract that exposes either a
+consuming choices to change it, or a choice that explicitly archives the contract and creates a new
+one.
+
+![consuming_choice](assets/consuming_choice.png)
+
+Contract ID's identify a contract uniquely. Every two contracts, even if they have the same field
+values and are of the same template, will have different contract ID's. If your application exposes
```
Comment (nemanja-da) [inline fundamental-concepts/contract-keys/1_chasing_contract_ids.md]:
```suggestion
values and are of the same template, will have different contract IDs. If your application exposes
```

File: fundamental-concepts/contract-keys/2_contract_keys.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    "key": "Alice",
+    "choice": "Follow",
+    "argument": {
+        "userToFollow": "Eve"
+}}' localhost:7575/v1/exercise
+```{{execute T1}}
+
+DAML allows you to specify one field of your template to be a [Contract Key](https://docs.daml.com/daml/reference/contract-keys.html).
+
+While the uniqueness of contract ID's is guaranteed by any DAML ledger, you as a DAML model writer
```
Comment (nemanja-da) [inline fundamental-concepts/contract-keys/2_contract_keys.md]:
```suggestion
While the uniqueness of contract IDs is guaranteed by any DAML ledger, you as a DAML model writer
```

File: fundamental-concepts/contract-keys/3_key_maintainers.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+```
+maintainer _2
+```
+
+Here the functions `_1: (a, b) -> a` and `_2: (a, b) -> b` are the projections onto the first or
+second factor of a tuple. 
+
+In the case of our `User` contract it is sensitive to assume that the `username` is a unique field
+among all `User` contracts. Knowing that every party sees the creation of a contract with the
+`username` field set to it's name, we make it the responsibility of such a party to ensure and check
```
Comment (nemanja-da) [inline fundamental-concepts/contract-keys/3_key_maintainers.md]:
```suggestion
`username` field set to its name, we make it the responsibility of such a party to ensure and check
```

File: fundamental-concepts/contract-keys/4_using_contract_keys.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+All DAML update actions that work on contract ID's also have a counter part for contract keys:
```
Comment (nemanja-da) [inline fundamental-concepts/contract-keys/4_using_contract_keys.md]:
```suggestion
All DAML update actions that work on contract IDs also have a counter part for contract keys:
```

File: fundamental-concepts/contract-keys/1_chasing_contract_ids.md (hunk 1 above)
Comment (ghost) [reply fundamental-concepts/contract-keys/1_chasing_contract_ids.md]: fixed.

File: fundamental-concepts/contract-keys/2_contract_keys.md (hunk 2 above)
Comment (ghost) [reply fundamental-concepts/contract-keys/2_contract_keys.md]: fixed.

File: fundamental-concepts/contract-keys/3_key_maintainers.md (hunk 3 above)
Comment (ghost) [reply fundamental-concepts/contract-keys/3_key_maintainers.md]: good catch, thanks!

File: fundamental-concepts/contract-keys/4_using_contract_keys.md (hunk 4 above)
Comment (ghost) [reply fundamental-concepts/contract-keys/4_using_contract_keys.md]: fixed.

**Pull request merged**